pkg/controller/logstorage/initializer/conditions_controller.go

```diff
@@ -90,18 +92,80 @@ func (r *LogStorageConditions) Reconcile(ctx context.Context, request reconcile.
+	}
+
+	// Map to keep track of the conditions status for each type.
+	statusMap := make(map[operatorv1.StatusConditionType]bool)
+
+	for _, logStorage := range logStorageInstances {
+		// Fetch TigeraStatus for the individual log storage subcontrollers.
+		ts := &operatorv1.TigeraStatus{}
+		if err := r.client.Get(ctx, types.NamespacedName{Name: logStorage}, ts); err != nil {
+			return reconcile.Result{}, err
+		}
+
+		// Update aggregated conditions based on fetched TigeraStatus
+		updateAggregatedConditions(ts.Status.Conditions, aggTigeraStatusConditions, statusMap, logStorage)
+	}
+
+	// Update tigera status conditions
+	switch {
+	case statusMap[operatorv1.ComponentDegraded]:
+		setAndClearTigeraStatus(aggTigeraStatusConditions, operatorv1.ComponentDegraded, string(operatorv1.ResourceDegraded), "")
+	case statusMap[operatorv1.ComponentProgressing]:
+		setAndClearTigeraStatus(aggTigeraStatusConditions, operatorv1.ComponentProgressing, string(operatorv1.ResourceProgressing), "")
+	case statusMap[operatorv1.ComponentAvailable]:
+		setAndClearTigeraStatus(aggTigeraStatusConditions, operatorv1.ComponentAvailable, string(operatorv1.AllObjectsAvailable), "All Objects are available")
 	}
 
-	ls.Status.Conditions = status.UpdateStatusCondition(ls.Status.Conditions, ts.Status.Conditions)
+	ls.Status.Conditions = status.UpdateStatusCondition(ls.Status.Conditions, aggTigeraStatusConditions)
 	if err := r.client.Status().Update(ctx, ls); err != nil {
 		log.WithValues("reason", err).Info("Failed to update LogStorage status conditions")
 		return reconcile.Result{}, err
 	}
 
 	return reconcile.Result{}, nil
 }
+
+func setAndClearTigeraStatus(aggTigeraStatusConditions []operatorv1.TigeraStatusCondition, conditionType operatorv1.StatusConditionType, reason, message string) {
+	for i := range aggTigeraStatusConditions {
+		if aggTigeraStatusConditions[i].Type == conditionType {
+			aggTigeraStatusConditions[i].Status = operatorv1.ConditionTrue
+		} else {
+			aggTigeraStatusConditions[i].Status = operatorv1.ConditionFalse
+			aggTigeraStatusConditions[i].Message = ""
+		}
+		aggTigeraStatusConditions[i].Reason = reason
+		if message != "" {
+			aggTigeraStatusConditions[i].Message = message
+		}
+	}
+}
+
+func updateAggregatedConditions(tsConditions []operatorv1.TigeraStatusCondition, aggTigeraStatusConditions []operatorv1.TigeraStatusCondition, statusMap map[operatorv1.StatusConditionType]bool, logStorageType string) {
+	for _, condition := range tsConditions {
+		for i := range aggTigeraStatusConditions {
+			if aggTigeraStatusConditions[i].Type == condition.Type {
+				aggTigeraStatusConditions[i].Status = condition.Status
+				aggTigeraStatusConditions[i].Message = fmt.Sprintf("%s%s for %s;", aggTigeraStatusConditions[i].Message, condition.Message, logStorageType)
+				if aggTigeraStatusConditions[i].LastTransitionTime.Time.Before(condition.LastTransitionTime.Time) {
```

Comment:
Done.
Updating the latest transition time here. However reverting it when there is no change in the object status in resetLastTransitionTimeStamp. So that i can compare the status and message once all the subcontrollers are aggregated.